Parsers/ASimNetworkSession/Parsers/vimNetworkSessionForcePointFirewall.yaml

```diff
@@ -0,0 +1,416 @@
+            "Connection timeout in state TCP_TIME_WAIT", "", "Success", "Timeout", "One end of the Connection acknowledged closing packet (FIN).",
+            "Connection timeout in state TCP_TIME_WAIT_ACK", "", "Failure",	"Timeout", "Waiting for ACK for the FIN status before going to time wait status (active close).",
+            "Connection timeout in state ICMP_ECHO", "", "Failure", "Timeout", "Ping reply is expected.",
+            "Connection timeout in state ICMP_REPLY_WAIT", "", "Failure", "Timeout", "Other ICMP request or reply types.",
+            "Connection was reset by client", "Reset Source", "Failure","Reset", "",
+            "Connection was reset by server", "Reset Destination", "Failure","Reset", "",
+            "invalid packet (CT)", "", "Failure", "Invalid TCP", "",
+            "not a (valid) SYN packet [A] (CT)", "", "Failure", "Invalid TCP", "",
+            "not a (valid) SYN packet [FA] (CT)", "", "Failure", "Invalid TCP", "",
+            "not a (valid) SYN packet [FPA] (CT)", "", "Failure", "Invalid TCP", "",
+            "not a (valid) SYN packet [PA] (CT)", "", "Failure", "Invalid TCP", "",
+            "not a (valid) SYN packet [RA] (CT)", "", "Failure", "Invalid TCP", "",
+            "not a (valid) SYN packet [SA] (CT)", "", "Failure", "Invalid TCP", "",
+            "TCP state violation","Deny","Failure", "Invalid TCP", "",
+            "TCP state violation: Connection end-point replied with ACK to SYN-packet. Connection refused.", "Deny", "Failure", "Invalid TCP", "",
+            "TSC error: Query timed out", "", "Failure", "Timeout", ""
+    ];
+    let parser = (
+        starttime:datetime=datetime(null), 
+        endtime:datetime=datetime(null), 
+        srcipaddr_has_any_prefix:dynamic=dynamic([]), 
+        dstipaddr_has_any_prefix:dynamic=dynamic([]), 
+        ipaddr_has_any_prefix:dynamic=dynamic([]),
+        dstportnumber:int=int(null), 
+        hostname_has_any:dynamic=dynamic([]), 
+        dvcaction:dynamic=dynamic([]), 
+        eventresult:string='*', 
+        disabled:bool=false) {
+    let src_or_any = set_union(srcipaddr_has_any_prefix, ipaddr_has_any_prefix); 
+    let dst_or_any = set_union(dstipaddr_has_any_prefix, ipaddr_has_any_prefix);
+    let ForcePointNetwork = CommonSecurityLog
+            | where not(disabled)
+            | where (isnull(starttime) or TimeGenerated >= starttime) 
+                and (isnull(endtime) or TimeGenerated <= endtime)
+            | where DeviceVendor=="FORCEPOINT" and DeviceProduct=="Firewall"
+            | where DeviceFacility in~ ("Inspection","Packet Filtering","File Filtering") and isnotempty(DeviceEventClassID) and DeviceEventClassID != "0"    
+            | where (isnull(dstportnumber) or (DestinationPort == dstportnumber) or (ApplicationProtocol has tostring(dstportnumber)))
+            | extend temp_isSrcMatch=has_any_ipv4_prefix(SourceIP,src_or_any), 
```

Comment:
Unfortunately, you can't do the filtering at the "ForcePointNetwork". It means you filter all traffic three times. It must be within each of the branches ("PacketFilteringData" etc.). You can copy the filtering part 3 times, or use an invoke function as I did [here](https://github.com/Azure/Azure-Sentinel/blob/master/Parsers/ASimAuthentication/Parsers/vimAuthenticationSshd.yaml)